serpentTools/parsers/results.py

```diff
@@ -1,15 +1,319 @@
+            raise SerpentToolsException("The repetitions of universes in the file is not equal.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        self._ChunkPosition()  # identify the 1st block for defining the dictionary
+        if self._ChunkPos == 0:
+            raise SerpentToolsException("The results file is corrupted.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        self._defineResultDict()  # Obtain the names of the variables to be stored
+        if self._resKeys == []:
+            raise SerpentToolsException("No variables appear in the results file.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+
+    def _defineResultDict(self):
+        """ Initializes the dictionary for ALL results included in .res.m"""
+        self._idxSect = [False, False, False]
+        self._resKeys = []  # Includes a complete set of .res variables
+        fObj = open(self.filePath, 'r')
+        numChar = 0  # count the number of characters
+        nUniv = len(self._univID)  # number of steps
+        nSteps = [self._univID[item] for item in self._univID][0]
+        for tline in fObj:
+            self._positionInFile(tline)
+            if numChar < self._ChunkPos[0]:
+                numChar += len(tline)
+                continue
+            # Not an empty line and not a comment
+            if tline.strip() != '' and not re.search(r'\%', tline):
+                currKey = re.search(r'^\w+', tline)
+                if currKey:
+                    keyName = convertVariableName(currKey.group())
+                    self._initResData(keyName, nSteps, tline)
+            numChar += len(tline)
+            if numChar >= self._ChunkPos[1]:
+                break
+        fObj.close()
+        self._idxSect = [False, False, False]
+
+    def _initResData(self, key, nSteps, tline):
+        """Initializes the dimensions of the variable"""
+        if key not in self._resKeys:
+            # Type of data (general meta, universes)
+            self._resKeys.append(key)
+            nvals = len(self._getDictVal(tline))
+            if self._idxSect[0]:
+                self.metadata[key] = ''
+            elif self._idxSect[1]:
+                self.resdata[key] = \
+                    empty((nSteps, nvals))
+
+    def _versionSerpent(self):
+        """Identifies the utilized version"""
+        with open(self.filePath, 'r') as fObject:
+            if fObject is None:
+                raise IOError("Attempting to read on file that does not exist.\n"
+                              "Parser: {}\nFile: {}".format(self, self.filePath))
+            for tline in fObject:
+                # If not an empty line or a comment
+                if re.search('VERSION', tline):
+                    version = self._getDictVal(tline)
+                    self._mapStrVersions(version)
+                    break
```

Comment:
Replace with `return`: more explicit that this function is done after finding `'VERSION'`, not just this `for` loop
Plus, you can raise a warning if you exit the `for` loop without finding `VERSION` if you use a `return` vs. a `break` here